exercises/paasio/tests/paasio.rs

```diff
@@ -0,0 +1,189 @@
+extern crate paasio;
+
+/// test a few read scenarios
+macro_rules! test_read {
+    ($(#[$attr:meta])* $modname:ident ($input:expr, $len:expr)) => {
+        mod $modname {
+            use std::io::{Read, BufReader};
+            use paasio::*;
+
+            const CHUNK_SIZE: usize = 2;
+
+            $(#[$attr])*
+            #[test]
+            fn test_read_passthrough() {
+                let data = $input;
+                let size = $len(&data);
+                let mut reader = ReadStats::new(data);
+
+                let mut buffer = Vec::with_capacity(size);
+                let qty_read = reader.read_to_end(&mut buffer);
+
+                assert!(qty_read.is_ok());
+                assert_eq!(size, qty_read.unwrap() );
```

Comment:
I agree with you, that would be nicer. So I tried it, and got this funky error:

```rust
   Compiling paasio v0.0.0 (file:///mnt/d/Users/coriolinus/Documents/Projects/exercism.io/xrust/exercises/paasio)
error[E0277]: can't compare `std::io::Error` with `std::io::Error`
   --> tests/paasio.rs:161:1
    |
161 |    test_read!(read_string (
    |   _^
    |  |_|
    | ||
162 | ||     "Twas brillig, and the slithy toves/Did gyre and gimble in the wabe:/All mimsy were the borogoves,/And the mome raths outgrabe.".as_bytes(),
163 | ||     |d: &[u8]| d.len()
164 | || ));
    | ||   ^
    | ||___|
    | |____no implementation for `std::io::Error == std::io::Error`
    |      in this macro invocation
    |
    = help: the trait `std::cmp::PartialEq` is not implemented for `std::io::Error`
    = note: required because of the requirements on the impl of `std::cmp::PartialEq` for `std::result::Result<usize, std::io::Error>`
    = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)
```

Apparently `std::io::Error` doesn't implement `PartialEq`, which makes that impossible. Oh well.